src/App.tsx

```diff
@@ -1,26 +1,471 @@
 /* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useRef, useState } from 'react';
+import cn from 'classnames';
 
-const USER_ID = 0;
+import * as todoService from './api/todos';
+import { Todo } from './types/Todo';
+import { LinkMode } from './types/LinkMode';
+import { EditingItem } from './types/EditingItem';
+import { CSSTransition, TransitionGroup } from 'react-transition-group';
+
+function getVisibleTodos(todos: Todo[], link: LinkMode) {
+  switch (link) {
+    case LinkMode.active:
+      return todos.filter(todo => !todo.completed);
+
+    case LinkMode.completed:
+      return todos.filter(todo => todo.completed);
+
+    default:
+      return todos;
+  }
+}
+
+function getTodosCount(todos: Todo[]) {
+  const all = todos.length;
+  const active = todos.reduce(
+    (sum, todo) => (todo.completed ? sum : sum + 1),
+    0,
+  );
+  const completed = all - active;
+
+  return { all, active, completed };
+}
+
+function getComplitedTodos(todos: Todo[]) {
+  return todos.filter(todo => todo.completed);
+}
+
+function verifyTitle(title: string) {
+  return title.trim() !== '';
+}
+
+function deleteItemFromArray(array: number[], item: number) {
+  const rest = [...array];
+  const index = rest.indexOf(item);
+
+  rest.splice(index, 1);
+
+  return rest;
+}
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const todosCount = getTodosCount(todos);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const [activeLink, setActiveLink] = useState(LinkMode.all);
+  const [title, setTitle] = useState('');
+  const [editingItem, setEditingItem] = useState<EditingItem>({
+    selectedTodo: null,
+    editedTitle: '',
+  });
+
+  const startEditing = (todo: Todo) => {
+    setEditingItem({ selectedTodo: todo, editedTitle: todo.title });
+  };
+
+  const stopEditing = () => {
+    setEditingItem({ selectedTodo: null, editedTitle: '' });
+  };
+
+  const updateEditingTitle = (editedTitle: string) => {
+    setEditingItem(prev => ({ ...prev, editedTitle }));
+  };
+
+  const [isSaving, setIsSaving] = useState(false);
+  const [deletingTodoIds, setDeletingTodoIds] = useState<number[]>([]);
+  const [updatingTodoIds, setUpdatingTodoIds] = useState<number[]>([]);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const timerId = useRef(0);
+
+  function focusInput() {
```

Comment:
You need to use 1 function creation type everywhere
```suggestion
  const focusInput = () => {
```